### PR TITLE
[PORT] [9.0] configuration helper

### PR DESCRIPTION
**Pull request by yvaucher**
Port to API 8.0 and make it compatible with a config definition in API
8.0. Remove some hacks with onchange and write to get and set the
related values.

### CODE REVIEW

**Comment by yvaucher** [comment]
@bealdav Could you review this ?

File: configuration_helper/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (yvaucher) [inline configuration_helper/models/__init__.py]: Missed one

File: configuration_helper/README.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+.. code-block:: python
+
+    from . company import ResCompany
+
+    class WhatiwantClassSettings(orm.TransientModel):
+        _inherit = ['res.config.settings', 'abstract.config.settings']
+        _name = 'whatiwant.config.settings'
+        # fields must be defined in ResCompany class
+        # related fields are automatically generated from previous definitions
+        _companyObject = ResCompany
```
Comment (bealdav) [inline configuration_helper/README.rst:34]: missing: 

`# all prefixed field with _prefix in res.company, will be available in 'whatiwant.config.settings' model`
`_prefix = 'prefixyouchoose_'`

File: configuration_helper/__init__.py [hunk 3]
```diff
@@ -1,22 +1,4 @@
...
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
-
-from . import config   # noqa
+# © 2014 David BEAL Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (bealdav) [inline configuration_helper/__init__.py]: you can remove lines above to save planet ;-)

File: configuration_helper/models/__init__.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
+# © 2014 David BEAL Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (bealdav) [inline configuration_helper/models/__init__.py]: 3 lines that can be removed

**Comment by bealdav** [comment]
Nice job increasing reliability of this module: test + code shorter
Thanks

File: configuration_helper/README.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+server-tools/issues/new?body=module:%20
+configuration_helper%0Aversion:%20
+9.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+
+Contributors
+------------
+
+* David BEAL <david.beal@akretion.com>
+* Sébastien BEAU <sebastien.beau@akretion.com>
+* Yannick Vaucher, Camptocamp, (code refactoring from his module 'delivery_carrier_label_postlogistics')
```
Comment (bealdav) [inline configuration_helper/README.rst]: please @yvaucher move your line in first position, it's more than a port: test, etc.

File: configuration_helper/README.rst (hunk 2 above)
Comment (yvaucher) [inline configuration_helper/README.rst:34]: Done

**Comment by yvaucher** [comment]
@bealdav Thanks for the review. I made the changes.

**Comment by bealdav** [comment]
:+1:

File: configuration_helper/models/config.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+
+class AbstractConfigSettings(models.AbstractModel):
+    _name = 'abstract.config.settings'
+    _description = 'Abstract configuration settings'
+    # prefix field name to differentiate fields in company with those in config
+    _prefix = 'setting_'
+    # this is the class name to import in your module
+    # (it should be ResCompany or res_company, depends of your code)
+    _companyObject = None
```
Comment (guewen) [inline configuration_helper/models/config.py:17]: Why a mixedCase suddenly?

Comment (guewen) [inline configuration_helper/models/config.py:17]: Ok, it was already there, so at least only to keep compatibility.

**Comment by guewen** [comment]
:+1:

File: configuration_helper/models/config.py (hunk 6 above)
Comment (bealdav) [inline configuration_helper/models/config.py:17]: You can break compatibility here to become cleaner, no problem between versions. 
There was no version for V8, only v7.

**Pull request merged**